__tests__/app.test.js

```diff
@@ -194,6 +194,48 @@ describe('/api/articles', () => {
     })
 })
 
+describe('/api/articles?topic', () => {
+    test('GET 200: Should return articles filtered by topic when passed an existing topic', () => {
+        return request(app)
+        .get('/api/articles?topic=cats')
+        .expect(200)
+        .then((response) => {
+            const { articles } = response.body
+            articles.forEach((article) => {
+                expect(article).toMatchObject({
+                    article_id: expect.any(Number),
+                    title: expect.any(String),
+                    topic: expect.any(String),
+                    author: expect.any(String),
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    article_img_url: expect.any(String),
+                    comment_count: expect.any(Number)
+                })
+                expect(article.topic).toBe('cats')
+            })
+        });
+    });
+    test('GET 200: Should return an empty array when passed a topic that does not exist', () => {
```

Comment:
Now if the topic exists but there are no articles, I wholeheartedly agree. Now if you're searching by a topic not in the topic table that should absolutely be a bad request.